pkgs/build-support/buildenv/builder.pl

```diff
@@ -123,7 +123,12 @@ sub findFiles {
 
     unless (-d $target && ($oldTarget eq "" || -d $oldTarget)) {
         if ($ignoreCollisions) {
-            warn "collision between `$target' and `$oldTarget'\n" if $ignoreCollisions == 1;
+            # Warn only if the contents differ. This reduces performance but prevents
+            # spamming users with false-positive warnings. It also allows developers
+            # to find real collisions that should be addressed.
+            if (!checkCollision($oldTarget, $target)) {
```

Comment:
This should only happen when checkCollisionContents is set to true

Reply:
Okay, I will update. This is the default so NixOS system-path would not even need to be changed.